cl/lib/elasticsearch_utils.py

```diff
@@ -2461,24 +2538,36 @@ def get_child_top_hits_limit(
      query hits.
     """
 
-    frontend_hits_limit = settings.CHILD_HITS_PER_RESULT
-    # Increase the CHILD_HITS_PER_RESULT value by 1. This is done to determine
-    # whether there are more than CHILD_HITS_PER_RESULT results, which would
+    match search_type:
+        case (
+            SEARCH_TYPES.RECAP
+            | SEARCH_TYPES.DOCKETS
+            | SEARCH_TYPES.RECAP_DOCUMENT
+        ):
+            child_limit = settings.RECAP_CHILD_HITS_PER_RESULT
+        case SEARCH_TYPES.OPINION:
+            child_limit = settings.OPINION_HITS_PER_RESULT
+        case _:
+            child_limit = 0
+
+    display_hits_limit = child_limit
+    # Increase the RECAP_CHILD_HITS_PER_RESULT value by 1. This is done to determine
+    # whether there are more than RECAP_CHILD_HITS_PER_RESULT results, which would
     # trigger the "View Additional Results" button on the frontend.
-    query_hits_limit = settings.CHILD_HITS_PER_RESULT + 1
+    query_hits_limit = child_limit + 1
 
     if search_type not in [SEARCH_TYPES.RECAP, SEARCH_TYPES.DOCKETS]:
-        return frontend_hits_limit, query_hits_limit
+        return display_hits_limit, query_hits_limit
 
     if search_type == SEARCH_TYPES.DOCKETS:
-        frontend_hits_limit = 1
+        display_hits_limit = 1
```

Comment:
We should refactor these if statements into the pattern matching block.

Reply:
Done!